cap/capture_panic_test.go

```diff
@@ -0,0 +1,85 @@
+package cap_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/capatazlib/go-capataz/cap"
+	. "github.com/capatazlib/go-capataz/internal/stest"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCapturePanic(t *testing.T) {
+	t.Run("one-level tree", func(t *testing.T) {
+		parentName := "root"
+		panicChild1, signalPanic1 := PanicOnSignalWorker(
+			1,
+			"child1",
+		)
+
+		events, err := ObserveSupervisor(
+			context.TODO(),
+			parentName,
+			cap.WithNodes(panicChild1),
+			[]cap.Opt{},
+			func(em EventManager) {
+				evIt := em.Iterator()
+				evIt.SkipTill(SupervisorStarted("root"))
+				signalPanic1(true /* done */)
+				evIt.SkipTill(WorkerStarted("root/child1"))
+			},
+		)
+
+		assert.NoError(t, err)
+
+		AssertExactMatch(t, events,
+			[]EventP{
+				// start children from left to right
+				WorkerStarted("root/child1"),
+				SupervisorStarted("root"),
+				WorkerFailedWith("root/child1", "Panicking child (1 out of 1)"),
```

Comment:
This is counter intuitive. The progression that makes sense to me is one of the following:

1. Worker attempts to start
2. panics while starting
3. fails to start
4. terminates

OR

1. worker attempts to start
2. worker successfully starts
3. while running business logic, it panics
4. terminates

The sequence you describe, with a failure, an apparent successful start, followed by an abrupt terminate, looks strange. Seems like you need a `WorkerFailedToStart()` predicate? Is that what you mean?

Reply:
I'll add comments explaining the sequence